pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -307,6 +312,69 @@ func createTargetConfigController(ctx context.Context, syncCtx factory.SyncConte
 	return false, nil
 }
 
+// setCloudControllerOwnerCondition sets the condition to False if either external cloud
+// provider has been successfully applied for all static pods or it's not set at all. Otherwise
+// it sets the condition to True.
+func setCloudControllerOwnerCondition(c TargetConfigController) error {
+	_, status, _, err := c.operatorClient.GetStaticPodOperatorState()
+	if err != nil {
+		return fmt.Errorf("could not get operator state: %v", err)
+	}
+	// Once we have successfully set CloudControllerOwner to False, we should not change it again.
+	if v1helpers.IsOperatorConditionFalse(status.Conditions, "CloudControllerOwner") {
+		return nil
+	}
+
+	// Get a config map for the latest revision. Then check if it contains "cloud-provider" argument which is equal
+	// to "external" or just empty.
+	latestRevisionConfigMapName := fmt.Sprintf("config-%d", status.LatestAvailableRevision)
+	cm, err := c.configMapLister.ConfigMaps(operatorclient.TargetNamespace).Get(latestRevisionConfigMapName)
```

Comment:
You should only reach for `config` config map, which will always reflect the latest rolled out revision content.

Reply:
Actually that was done intentionally to prevent a race condition in KCMO. Imagine a situation:
There is revision 10 which doesn't have cloud-provider set to external. KCM object status has "10" in latestAvailableRevision. NodeInstallerProgressingCondition is false, the message is "AllNodesAtLatestRevision".

Then we apply the FeatureGate to enable CCMs. New revision "11" is created. "config" cm starts to point to "config-11". But latestAvailableRevision for KCM is still "10", and the condition doesn't change for some short period of time. A second later the controller will update latestAvailableRevision to "11", but it may be too late, because we've already set CloudControllerOwner to False.

So, if we fetch just "config", we may mistakenly think that the right configuration has already been applied, because KCMO will report AllNodesAtLatestRevision, which might not be true.

To avoid this race condition we take into consideration the LatestAvailableRevision value and fetch the config map with this number as a suffix.